src/mame/drivers/photoplysx.cpp

```diff
@@ -0,0 +1,103 @@
+// license:BSD-3-Clause
+// copyright-holders:
+/*
+	Funworld Photo Play Spirit Xtreme
+
+	CPU: Intel CELERON 2 GHz / 128 / 400 (SL6VR)
+	RAM: 256MB DDR 333
+	PCB: Intel 865G + Intel Extreme Graphics 2 + Intel FW82801FR
+	I/O: Cypress AN2131SC (12.000 MHz xtal) + Winbond W83627HF + Realtek RTL8101L
+	BIOS: 03/11/2009-I865G-6A79AD4EC-00 (Pm49FL004T-33JC)
+	Dongle: USB
+	Sound: C-Media CMI9761A
+*/
+
+#include "emu.h"
+#include "cpu/i386/i386.h"
+#include "screen.h"
+
+class photoplaysx_state : public driver_device
+{
+public:
+	photoplaysx_state(const machine_config &mconfig, device_type type, const char *tag)
+		: driver_device(mconfig, type, tag),
+		m_maincpu(*this, "maincpu")
+	{ }
+
+	void photoplaysx(machine_config &config);
+
+private:
+	required_device<cpu_device> m_maincpu;
+
+	virtual void machine_start() override;
+	virtual void machine_reset() override;
+	virtual void video_start() override;

```

Comment:
Overridden protected members should be protected.

Reply:
Fixed with PR #7355